reactive/etcd.py

```diff
@@ -821,6 +834,78 @@ def remove_nrpe_config(nagios=None):
         nrpe_setup.remove_check(shortname=service)
 
 
+@when('endpoint.prometheus.joined',
+      'leadership.is_leader',
+      'certificates.ca.available')
+def register_prometheus_jobs():
+    log('Registering Prometheus metrics collection.')
+    prometheus = endpoint_from_flag('endpoint.prometheus.joined')
+    cluster = endpoint_from_flag('cluster.joined')
+
+    if not etcd_reachable_from_endpoint(prometheus.endpoint_name):
+        log('Aborting Prometheus metrics collection. Etcd is not reachable by'
+            ' prometheus client', WARNING)
+        return
+
+    peer_ips = cluster.get_db_ingress_addresses() if cluster else []
+    peer_ips.append(get_ingress_address('db'))
+    targets = ["{}:{}".format(ip, config('port')) for ip in peer_ips]
+    log('Configuring Prometheus scrape targets: {}'.format(targets), DEBUG)
+    prometheus.register_job(job_name='etcd',
+                            job_data={
+                                'scheme': 'https',
+                                'static_configs': [
+                                    {'targets': targets},
+                                ]
+                            })
+    set_flag('prometheus.configured')
+
+
+@when('endpoint.prometheus.departed')
+def reset_prometheus_config():
+    clear_flag('prometheus.configured')
+    clear_flag('grafana.configured')
+
+
+@when(
+    "prometheus.configured",
+    "endpoint.grafana.joined",
+    "leadership.is_leader"
+)
+@when_not("grafana.configured")
+def register_grafana_dashboard():
+    log("Configuring grafana dashboard", level=hookenv.INFO)
+    grafana = endpoint_from_flag("endpoint.grafana.joined")
+    prometheus = endpoint_from_flag('endpoint.prometheus.joined')
+
+    if not prometheus:
+        log(
+            "Prometheus relation not available. Skipping Grafana"
+            " configuration.", hookenv.WARNING)
+        return
+
+    if len(prometheus.relations) > 1:
+        log(
+            "Multiple prometheus relations detected. Default Grafana dashboard"
+            " will configure only with one of them as datasource.",
+            hookenv.WARNING)
+
+    datasource = prometheus.relations[0].application_name
+    dashboard = render_grafana_dashboard(datasource)
+
+    log("Rendered Grafana dashboard:\n{}".format(json.dumps(dashboard)),
+        level=hookenv.DEBUG)
+    grafana.register_dashboard(name=GRAFANA_DASHBOARD_NAME,
+                               dashboard=dashboard)
+    log('Grafana dashboard "{}" registered.'.format(GRAFANA_DASHBOARD_NAME))
+    set_flag("grafana.configured")
+
+
+@when("endpoint.grafana.departed", "grafana.configured")
+def unregister_grafana_dashboard():
+    clear_flag("grafana.configured")
```

Comment:
Same comment as above.

```suggestion
register_trigger(when_not="endpoint.grafana.joined", clear_flag="grafana.configured")
```